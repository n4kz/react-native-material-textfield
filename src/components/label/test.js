import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Label from '.';

/* eslint-env jest */

const text = 'test';
const props = {
  baseSize: 32,
  basePadding: 4,
  fontSize: 16,
  activeFontSize: 12,
  tintColor: 'blue',
  baseColor: 'black',
  errorColor: 'red',
  animationDuration: 225,
};

it('renders label', () => {
  let label = renderer
    .create(<Label {...props}>{text}</Label>);

  expect(label.toJSON())
    .toMatchSnapshot();
});

it('renders active label', () => {
  let label = renderer
    .create(<Label active {...props}>{text}</Label>);

  expect(label.toJSON())
    .toMatchSnapshot();
});

it('renders focused label', () => {
  let label = renderer
    .create(<Label focused {...props}>{text}</Label>);

  expect(label.toJSON())
    .toMatchSnapshot();
});

it('renders errored label', () => {
  let label = renderer
    .create(<Label errored {...props}>{text}</Label>);

  expect(label.toJSON())
    .toMatchSnapshot();
});

it('renders restricted label', () => {
  let label = renderer
    .create(<Label restricted {...props}>{text}</Label>);

  expect(label.toJSON())
    .toMatchSnapshot();
});
