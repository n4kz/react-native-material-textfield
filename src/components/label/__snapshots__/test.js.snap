// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders active label 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "position": "absolute",
      "top": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    collapsable={undefined}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0, 0, 0, 1)",
        "fontSize": 12,
      }
    }
  >
    test
  </Text>
</View>
`;

exports[`renders errored label 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "position": "absolute",
      "top": 36,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    collapsable={undefined}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(255, 0, 0, 1)",
        "fontSize": 16,
      }
    }
  >
    test
  </Text>
</View>
`;

exports[`renders focused label 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "position": "absolute",
      "top": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    collapsable={undefined}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0, 0, 255, 1)",
        "fontSize": 12,
      }
    }
  >
    test
  </Text>
</View>
`;

exports[`renders label 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "position": "absolute",
      "top": 36,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    collapsable={undefined}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0, 0, 0, 1)",
        "fontSize": 16,
      }
    }
  >
    test
  </Text>
</View>
`;

exports[`renders restricted label 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "position": "absolute",
      "top": 36,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    collapsable={undefined}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "color": "red",
        "fontSize": 16,
      }
    }
  >
    test
  </Text>
</View>
`;
